beamer/tests/agent/test_challenge.py

```diff
@@ -93,21 +105,31 @@ def test_challenge_2(request_manager, token, config):
         request_manager.challengeClaim(claim.claimId, sender=charlie, value=claim.claimerStake + 1)
 
         # Charlie's claim.
-        claim = collector.next_event()
-        assert claim is not None
-        assert claim.claimerStake < claim.challengerStakeTotal
+        claim_event = collector.next_event()
+        assert claim_event is not None
+        assert claim_event.claimerStake < claim_event.challengerStakeTotal
 
-        # Bob's claim.
-        claim = collector.next_event()
-        assert claim is not None
-        assert claim.claimerStake > claim.challengerStakeTotal
+        with Sleeper(5) as sleeper:
+            while (claim := agent.get_context(direction).claims.get(claim_event.claimId)) is None:
+                sleeper.sleep(0.1)
 
-        ape.chain.mine(timestamp=claim.termination)
-        request_manager.withdraw(claim.claimId, sender=agent.address)
+        with Sleeper(5) as sleeper:
+            while (request := agent.get_context(direction).requests.get(request_id)) is None:
+                sleeper.sleep(0.1)
+
+        with freeze_time(
```

Comment:
I assume if we use the chain's timestamp for determining the L1 finalization, we don't need this context manager here, do we?

Reply:
same here, as we continue like before i will keep this.